featurehub-react-sdk/src/components/FeatureHub.tsx

```diff
@@ -0,0 +1,86 @@
+import {
+  ClientContext,
+  EdgeFeatureHubConfig,
+  FeatureHubPollingClient,
+  Readyness,
+} from "featurehub-javascript-client-sdk"
+
+export const FeatureHubContext = createContext<ClientContext | null>(null)
+FeatureHubContext.displayName = "FeatureHub"
+
+type Props = {
+  /** The url to the running instance of the FeatureHub EDGE API */
+  readonly url: string
+  /** The FeatureHub API key -- can be found in the FeatureHub Admin Console */
+  readonly apiKey: string
+  /** Scopes FeatureHub context to username -- otherwise, context will be anonymous. */
+  readonly username?: string
+  /** Interval (in milliseconds) to poll FeatureHub for updates. [default: 60 seconds] */
+  readonly pollInterval?: number
+  /** The React application tree to inject the FeatureHub client into */
+  readonly children: JSX.Element
+}
+
+/**
+ * The FeatureHub React component. Use this as a top-level wrapper around your
+ * main App component.
+ * 
+ * @param {string} url - the url of the FeatureHub instance (required)
+ * @param {string} apiKey - the apiKey key to use. Make sure it is the server-eval key! (required)
+ * @param {string} username - the optional username to add user information to the FeatureHub context (optional)
+ * @param {number} pollInterval - the desired polling interval (ms) to check for value updates (optional -- default 60 seconds)
+ * @param {JSX} children - the React component tree to inject FeatureHub into (required)
+ * 
+ */
+export default function FeatureHub({
+  url,
+  apiKey,
+  username,
+  pollInterval = 60000,
+  children,
+}: Props): JSX.Element {
+  const fhConfig = useMemo(() => new EdgeFeatureHubConfig(url, apiKey), [apiKey, url])
+  const [client, setClient] = useState(fhConfig.newContext())
+  const listener = useCallback(async (readyness: Readyness) => {
+      const ready = readyness === Readyness.Ready
+
+      if (!ready) {
+        console.warn("Cannot connect to FeatureHub!")
+        return
+      }
+
+      console.info("FeatureHub React SDK connected!");
+
+      if (!username) {
+        console.info("Username not found -- using anonymous user context.")
+        return
+      }
+
+      console.info("Username found -- build new context with userKey.")
+      setClient(await fhConfig.newContext().userKey(username).build())
+  }, [username])
+
+  useEffect(() => {
+    fhConfig.edgeServiceProvider((repo, c) => new FeatureHubPollingClient(repo, c, pollInterval))
+  }, [pollInterval])
+
+  useEffect(() => {
+    // Remove potential previous existing listener (for use-case when the username updates while component still mounted)
+    fhConfig.removeReadinessListener(listener)
+    fhConfig.addReadinessListener(listener, true)
+    fhConfig.init()
+
+    return () => {
+      console.info("FeatureHub unmounting -- terminating connection!")
+      fhConfig.removeReadinessListener(listener)
+      fhConfig.close()
+    }
+  }, [username])
```

Comment:
What's happening here is probably worth talking about because I don't think it's correct (specifically the `removeReadinessListener` cleanup/guard we're doing on line 70).

The intent is to clear the previous existing listener that was originally added if a new one gets generated as a result of the `username` changing -- I don't believe it's doing that properly and I have a suspicion that it will still keep the old one around thus leaving us with two active readiness listeners when we only want one.